test/integration/test_download.py

```diff
@@ -79,3 +77,35 @@ def test_small_unverified(self):
         bucket = self.create_bucket()
         f = self._file_helper(bucket, sha1_sum='do_not_verify')
         assert not f.download_version.content_sha1_verified
+
+    def test_gzip(self):
+        bucket = self.create_bucket()
+        with TempDir() as temp_dir:
+            temp_dir = pathlib.Path(temp_dir)
+            source_file = temp_dir / 'compressed_file.gz'
+            downloaded_uncompressed_file = temp_dir / 'downloaded_uncompressed_file'
+            downloaded_compressed_file = temp_dir / 'downloaded_compressed_file'
+
+            data_to_write = b"I'm about to be compressed and sent to the cloud, yay!\n" * 100  # too short files failed somehow
+            with gzip.open(source_file, 'wb') as gzip_file:
+                gzip_file.write(data_to_write)
+            file_version = bucket.upload_local_file(
+                str(source_file), 'gzipped_file', file_infos={'b2-content-encoding': 'gzip'}
```

Comment:
I think this was supposed to use `X-Bz-Info-b2-content-encoding` header, not file_info dict